From a8a0d2ab82729fb79302453ff38dc745f51b089a Mon Sep 17 00:00:00 2001
From: Antony Abee Prakash XV <antonyabee.prakashxv@adlinktech.com>
Date: Fri, 31 Jul 2020 11:05:02 +0530
Subject: [PATCH 17/17] Add-LVDS-support

- Add support for single channel LVDS panel HSD100IFW1 (Hannstar)
  of resolution 1024x600
- Change dcic2 (Display Content Integrity Checker) to watch LVDS0 instead
  of LVDS1.
- Make LVDS0 the primary port and remove LVDS1 from the LDB (LVDS Display
  Bridge) configurations.
- Move fb 'status' to the common .dtsi, where all devices are enabled.

Signed-off-by: Antony Abee Prakash XV <antonyabee.prakashxv@adlinktech.com>
---
 arch/arm/boot/dts/lec-imx6.dtsi  | 56 +++++++-------------------------
 arch/arm/boot/dts/lec-imx6q.dts  | 12 -------
 arch/arm/boot/dts/lec-imx6qp.dts |  5 ---
 arch/arm/boot/dts/lec-imx6s.dts  |  4 ---
 4 files changed, 12 insertions(+), 65 deletions(-)

diff --git a/arch/arm/boot/dts/lec-imx6.dtsi b/arch/arm/boot/dts/lec-imx6.dtsi
index 86bd50621377..000a7624fa2b 100644
--- a/arch/arm/boot/dts/lec-imx6.dtsi
+++ b/arch/arm/boot/dts/lec-imx6.dtsi
@@ -16,7 +16,6 @@
 		mxcfb0 = &mxcfb1;
 		mxcfb1 = &mxcfb2;
 		mxcfb2 = &mxcfb3;
-		mxcfb3 = &mxcfb4;
 	};
 
 	chosen {
@@ -234,17 +233,6 @@
 	};
 
 	mxcfb2: fb@1 {
-		compatible = "fsl,mxc_sdc_fb";
-		disp_dev = "hdmi";
-		interface_pix_fmt = "RGB24";
-		mode_str ="1920x1080M@60";
-		default_bpp = <24>;
-		int_clk = <0>;
-		late_init = <0>;
-		status = "disabled";
-	};
-
-	mxcfb3: fb@2 {
 		compatible = "fsl,mxc_sdc_fb";
 		disp_dev = "lcd";
 		interface_pix_fmt = "RGB565";
@@ -255,14 +243,15 @@
 		status = "disabled";
 	};
 
-	mxcfb4: fb@3 {
+	mxcfb3: fb@2 {	/* IPU2 DI0: /dev/fb3=LVDS, fb4=IPU2's overlay
+			* Solo/DL: IPU1, fb2 competing with par. RGB, no overlay */
 		compatible = "fsl,mxc_sdc_fb";
 		disp_dev = "ldb";
-		interface_pix_fmt = "RGB666";
-		default_bpp = <16>;
+		interface_pix_fmt = "RGB24";
+		default_bpp = <32>;
 		int_clk = <0>;
 		late_init = <0>;
-		status = "disabled";
+		status = "okay";
 	};
 
 	lcd@0 {
@@ -376,8 +365,8 @@
 #endif
 
 &clks {
-	assigned-clocks = <&clks IMX6QDL_CLK_LDB_DI0_SEL>,
-			  <&clks IMX6QDL_CLK_LDB_DI1_SEL>;
+	assigned-clocks = <&clks IMX6QDL_CLK_LDB_DI0_SEL>,	/*LVDS*/
+			  <&clks IMX6QDL_CLK_LDB_DI1_SEL>;	/*unused*/
 	assigned-clock-parents = <&clks IMX6QDL_CLK_PLL2_PFD0_352M>,
 				 <&clks IMX6QDL_CLK_PLL2_PFD0_352M>;
 };
@@ -390,7 +379,7 @@
 
 &dcic2 {
 	dcic_id = <1>;
-	dcic_mux = "dcic-lvds1";
+	dcic_mux = "dcic-lvds0";
 	status = "okay";
 };
 
@@ -1198,8 +1187,9 @@
 	status = "okay";
 
 	lvds-channel@0 {
-		fsl,data-mapping = "spwg";
-		fsl,data-width = <18>;
+		fsl,data-mapping = "spwg";	/* spwg or jeida (see chapter "Bit Mapping" in i.MX6 Reference Manual) */
+		fsl,data-width = <24>;
+		primary;
 		status = "okay";
 
 		display-timings {
@@ -1207,29 +1197,7 @@
 			timing0: hsd100pxn1 {
 				clock-frequency = <65000000>;
 				hactive = <1024>;
-				vactive = <768>;
-				hback-porch = <220>;
-				hfront-porch = <40>;
-				vback-porch = <21>;
-				vfront-porch = <7>;
-				hsync-len = <60>;
-				vsync-len = <10>;
-			};
-		};
-	};
-
-	lvds-channel@1 {
-		fsl,data-mapping = "spwg";
-		fsl,data-width = <18>;
-		primary;
-		status = "okay";
-
-		display-timings {
-			native-mode = <&timing1>;
-			timing1: hsd100pxn1 {
-				clock-frequency = <65000000>;
-				hactive = <1024>;
-				vactive = <768>;
+				vactive = <600>;
 				hback-porch = <220>;
 				hfront-porch = <40>;
 				vback-porch = <21>;
diff --git a/arch/arm/boot/dts/lec-imx6q.dts b/arch/arm/boot/dts/lec-imx6q.dts
index fb0d1377df95..a3903d24086f 100644
--- a/arch/arm/boot/dts/lec-imx6q.dts
+++ b/arch/arm/boot/dts/lec-imx6q.dts
@@ -23,24 +23,12 @@
 	lvds-channel@0 {
 		crtc = "ipu2-di0";
 	};
-
-	lvds-channel@1 {
-		crtc = "ipu2-di1";
-	};
 };
 
 &mxcfb2 {
 	status = "okay";
 };
 
-&mxcfb3 {
-	status = "okay";
-};
-
-&mxcfb4 {
-	status = "okay";
-};
-
 &sata {
 	/* Freescale's defaults */
 	fsl,transmit-level-mV = <1025>;
diff --git a/arch/arm/boot/dts/lec-imx6qp.dts b/arch/arm/boot/dts/lec-imx6qp.dts
index 3639b1bc455b..60658261da37 100644
--- a/arch/arm/boot/dts/lec-imx6qp.dts
+++ b/arch/arm/boot/dts/lec-imx6qp.dts
@@ -109,11 +109,6 @@
 	status = "okay";
 };
 
-&mxcfb4 {
-	prefetch;
-	status = "okay";
-};
-
 &ldb {
 	lvds-channel@0 {
 		crtc = "ipu2-di0";
diff --git a/arch/arm/boot/dts/lec-imx6s.dts b/arch/arm/boot/dts/lec-imx6s.dts
index 8bdcbb844855..e8609c1161e1 100644
--- a/arch/arm/boot/dts/lec-imx6s.dts
+++ b/arch/arm/boot/dts/lec-imx6s.dts
@@ -155,10 +155,6 @@
 	lvds-channel@0 {
 		crtc = "ipu1-di0";
 	};
-
-	lvds-channel@1 {
-		crtc = "ipu1-di1";
-	};
 };
 
 &mxcfb2 {
-- 
2.17.1

